### ISSUE TITLE
Consolidate string compare throughout

### ISSUE THREAD

**Issue by mhofman** [Agoric/Agoric#10335]
## Describe the bug

We have different ways of comparing strings, none of which are "correct". Endo will soon include https://github.com/endojs/endo/pull/2008, meaning we can use its `compareByCodePoints` throughout.

## To Reproduce

Incorrect string comparison
- `a < b` or any variation. This results in a comparison by UTF-16 code units.
- `[a, b].sort()` similarly uses code units
- `a.localeCompare()` compares by environment specific local rules. The SES shim is supposed to [tame this by default](https://github.com/endojs/endo/blob/fad5c68a534193df6af86d53095b1257542c33e6/packages/ses/src/tame-locale-methods.js#L46), but I have seen surprising results (see https://github.com/Agoric/agoric-sdk/pull/10299 test and undo localeCompare change).

## Expected behavior
Comparison by codepoints which we're trying to standardize on, and is the behavior for any implementation that compares utf-8 encoded bytes like SQLite, golang (doesn't strictly requires that string be utf-8, but literals and range iterations are), and JSON (which specifies utf-8 as the default encoding).

## Additional context

https://github.com/Agoric/agoric-sdk/pull/10299 fixes the more egregious of these cases, which caused a discrepancy in the order of Stores when they're heap or virtual/durable, and whether they're in a real or fake liveslots environment. In [comments](https://github.com/Agoric/agoric-sdk/pull/10299#discussion_r1811208633), @gibson042 also points out at other cases of mistaken comparisons (likely not exhaustive).

**Comment by mhofman** [comment]
I have recently verified that no durable data contains codepoint in the problematic range (0xE000-0xFFFF), meaning we can safely change the sorting used in endo to use a codepoint compare.

FYI @erights

**Comment by erights** [comment]
> Now that's we've updated to a version of endo with https://github.com/endojs/endo/pull/2008,

No, https://github.com/endojs/endo/pull/2008 is still in draft. We've only decided to proceed with it. We have not yet done so.

**Comment by erights** [comment]
Oh sorry, the quote I responded to is old. Regarding the recent message above: +1

**Comment by mhofman** [comment]
Yeah I must have written the original issue assuming it was getting merged...
